static/app/views/automations/components/ruleNode.tsx

```diff
@@ -0,0 +1,91 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/core/button';
+import {IconDelete} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {DataCondition} from 'sentry/types/workflowEngine/dataConditions';
+import {
+  DataConditionNodeContext,
+  dataConditionNodesMap,
+  useDataConditionNodeContext,
+} from 'sentry/views/automations/components/dataConditionNodes';
+
+interface RuleNodeProps {
+  condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>;
```

Comment:
If this is a common thing you're doing for the automation builder, maybe we should split the `DataCondition` type into a `NewDataCondition` without these and a new `DataCondition` that has them, but is readonly?

Reply:
updated the `DataCondition` types here in https://github.com/getsentry/sentry/commit/f228deee96773447ca29d0b548c6d9433ff29b6c

https://github.com/getsentry/sentry/blob/f228deee96773447ca29d0b548c6d9433ff29b6c/static/app/types/workflowEngine/dataConditions.tsx#L70-L82

just want to call out that i made `NewDataCondition.dataConditionGroup` optional since not all data conditions have additional nesting, and switched `DataConditionGroup` to use `NewDataConditionGroup`